lib/src/recovery_protocol/angular_analyzer_protocol.dart

```diff
@@ -0,0 +1,602 @@
+          ? current.quoteOffset
+          : current.offset;
+      returnToken = new NgToken.generateErrorSynthetic(
+          _offset, NgTokenType.beforeElementDecorator,
+          lexeme: " ");
+      returnState = NgScannerState.scanElementDecorator;
+    } else if (type == NgSimpleTokenType.commentBegin ||
+        type == NgSimpleTokenType.openTagStart ||
+        type == NgSimpleTokenType.closeTagStart ||
+        type == NgSimpleTokenType.EOF) {
+      reader.putBack(current);
+      returnToken = new NgToken.generateErrorSynthetic(
+          current.offset, NgTokenType.openElementEnd);
+      returnState = NgScannerState.scanStart;
+    }
+
+    return new RecoverySolution(returnState, returnToken);
+  }
+
+  @override
+  RecoverySolution scanAfterInterpolation(
+      NgSimpleToken current, NgTokenReversibleReader reader) {
+    if (current.type == NgSimpleTokenType.EOF) {
+      reader.putBack(current);
+      return new RecoverySolution(
+          NgScannerState.scanStart,
+          new NgToken.generateErrorSynthetic(
+              current.offset, NgTokenType.interpolationEnd));
+    }
+    return new RecoverySolution.skip();
+  }
+
+  @override
+  RecoverySolution scanBeforeElementDecorator(
+      NgSimpleToken current, NgTokenReversibleReader reader) {
+    return new RecoverySolution.skip();
+  }
+
+  @override
+  RecoverySolution scanBeforeInterpolation(
+      NgSimpleToken current, NgTokenReversibleReader reader) {
+    return new RecoverySolution.skip();
+  }
+
+  @override
+  RecoverySolution scanCloseElementEnd(
+      NgSimpleToken current, NgTokenReversibleReader reader) {
+    return new RecoverySolution.skip();
+  }
+
+  @override
+  RecoverySolution scanComment(
+      NgSimpleToken current, NgTokenReversibleReader reader) {
+    if (current.type == NgSimpleTokenType.EOF) {
+      return new RecoverySolution(
+          NgScannerState.scanStart,
+          new NgToken.generateErrorSynthetic(
+              current.offset, NgTokenType.commentEnd));
+    }
+    return new RecoverySolution.skip();
+  }
+
+  @override
+  RecoverySolution scanInterpolation(
+      NgSimpleToken current, NgTokenReversibleReader reader) {
+    return new RecoverySolution.skip();
+  }
+
+  @override
+  RecoverySolution scanElementDecorator(
+      NgSimpleToken current, NgTokenReversibleReader reader) {
+    NgScannerState returnState;
+    NgToken returnToken;
+    NgSimpleTokenType type = current.type;
+    int offset = current.offset;
+
+    if (type == NgSimpleTokenType.equalSign ||
+        type == NgSimpleTokenType.commentBegin ||
+        type == NgSimpleTokenType.openTagStart ||
+        type == NgSimpleTokenType.closeTagStart ||
+        type == NgSimpleTokenType.EOF ||
+        type == NgSimpleTokenType.doubleQuote ||
+        type == NgSimpleTokenType.singleQuote) {
+      reader.putBack(current);
+      int _offset = (current is NgSimpleQuoteToken)
+          ? current.quoteOffset
+          : current.offset;
```

Comment:
Yeah I think it would be better that way too. I'll probably switch it with the next reformatting/fixing.